### PR TITLE
update fave87h

### PR DESCRIPTION
**Pull request by duilchoi**
## Description

update fave87h RGB index,  layout.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/linworks/fave87h/info.json [hunk 1]
```diff
@@ -24,9 +24,6 @@
     "layout_aliases": {
         "LAYOUT": "LAYOUT_tkl_ansi_split_bs"
     },
-    "community_layouts": [
```
Comment (drashna) [inline keyboards/linworks/fave87h/info.json:27]: Is there a reason for removing this?

Comment (duilchoi) [reply keyboards/linworks/fave87h/info.json:27]: sorry....It's a mistake.update done!.

File: keyboards/linworks/fave87h/fave87h.c [hunk 2]
```diff
@@ -34,15 +34,28 @@ led_config_t g_led_config = {
...
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 4, 4, 4, 4,
+        4, 4, 4, 4, 4, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 4, 1,
+        1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 
+        1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1
```
Comment (zvecr) [inline keyboards/linworks/fave87h/fave87h.c:58]: Virtually everything being set a a modifier (`1`) is incorrect.

File: keyboards/linworks/fave87h/info.json [hunk 3]
```diff
@@ -64,7 +64,7 @@
                 {"matrix": [1, 10], "x": 10, "y": 1.25},
                 {"matrix": [1, 11], "x": 11, "y": 1.25},
                 {"matrix": [1, 12], "x": 12, "y": 1.25},
-                {"matrix": [1, 13], "x": 13, "y": 1.25, "w": 2},
+                {"matrix": [1, 13], "x": 13, "y": 1.25, "w":2},
```
Comment (zvecr) [inline keyboards/linworks/fave87h/info.json:67]:
```suggestion
                {"matrix": [1, 13], "x": 13, "y": 1.25, "w": 2},
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave87h/fave87h.c (hunk 2 above)
Comment (duilchoi) [reply keyboards/linworks/fave87h/fave87h.c:58]: Updated from 1 to 2.

File: keyboards/linworks/fave87h/info.json
Comment (dunk2k) [inline keyboards/linworks/fave87h/info.json:1]: Could you revert removal of line 172 and addition of line 209 please?

`"x"` and `"y"` sequence take priority over matrix co-ordinate sequence

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave87h/info.json [hunk 4]
```diff
@@ -237,5 +238,6 @@
                 {"matrix": [5, 16], "x": 17.25, "y": 5.25}
             ]
         }
+
```
Comment (dunk2k) [inline keyboards/linworks/fave87h/info.json:241]:
```suggestion
```

File: keyboards/linworks/fave87h/info.json [hunk 5]
```diff
@@ -132,6 +132,7 @@
                 {"matrix": [5, 16], "x": 17.25, "y": 5.25}
             ]
         },
+
```
Comment (dunk2k) [inline keyboards/linworks/fave87h/info.json:135]:
```suggestion
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave87h/info.json
Comment (duilchoi) [reply keyboards/linworks/fave87h/info.json:1]: Do you mean add line 209 after line 172?

Comment (dunk2k) [reply keyboards/linworks/fave87h/info.json:1]: If that's how you interpret my suggestion, then yes 🙂

`"x"` and `"y"` sequence take priority over matrix co-ordinate sequence.

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by dunk2k — APPROVED** [review]

File: keyboards/linworks/fave87h/keymaps/default/keymap.c [hunk 6]
```diff
@@ -22,21 +22,39 @@ enum layers {
...
+        KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [_LAYER3] = LAYOUT(
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                             KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS
```
Comment (drashna) [inline keyboards/linworks/fave87h/keymaps/default/keymap.c:57]: PLease remove the empty layers.
```suggestion
```

File: keyboards/linworks/fave87h/keymaps/via/keymap.c [hunk 7]
```diff
@@ -17,46 +17,44 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
...
-        KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,                             KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,                            KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,           KC_TRNS, KC_TRNS, KC_TRNS
     )
```
Comment (drashna) [inline keyboards/linworks/fave87h/keymaps/via/keymap.c:58]: These appear to be empty layers.  Please remove (the dynamic keymap feature auto-populates layers with transparency if it's  not explicitly defined, so having empty layers just artificially inflates the compile size).
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/linworks/fave87h/fave87h.c [hunk 8]
```diff
@@ -34,15 +34,28 @@ led_config_t g_led_config = {
...
-        1, 1, 1,       4,          1, 1,    1, 1, 1, 1, 1
+        4, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 4, 4, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 4, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 1, 4, 4, 4, 4,
+        4, 4, 4, 4, 4, 1, 1, 1, 1, 1,
+        1, 1, 1, 1, 1, 4, 1,
```
Comment (zvecr) [inline keyboards/linworks/fave87h/fave87h.c:54]: These are still incorrect as previously mentioned. The alphanumeric keys are not indicators.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave87h/fave87h.c (hunk 8 above)
Comment (drashna) [reply keyboards/linworks/fave87h/fave87h.c:54]:
```suggestion
        1,    4, 4, 4, 4, 1, 1, 1, 1, 4, 4, 4, 4, 1, 1, 1,
        1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 1, 1, 1,
        1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1, 1, 1, 1,
        9, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1,
        1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1,       4,
        1, 1, 1,       4,          1, 1,    1, 1, 1, 1, 1
```

File: keyboards/linworks/fave87h/fave87h.c [hunk 9]
```diff
@@ -43,6 +49,10 @@ led_config_t g_led_config = {
         9, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1,
         1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,    1,       4,
         1, 1, 1,       4,          1, 1,    1, 1, 1, 1, 1
```
Comment (drashna) [inline keyboards/linworks/fave87h/fave87h.c:51]: Sorry, missed this: 
```suggestion
        1, 1, 1,       4,          1, 1,    1, 1, 1, 1, 1,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Keyboard updates need to go through `develop`.

File: keyboards/linworks/fave87h/config.h [hunk 10]
```diff
@@ -18,7 +18,7 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 /* Define RGB */
-#define RGB_MATRIX_LED_COUNT 87
+#define RGB_MATRIX_LED_COUNT 127
```
Comment (fauxpark) [inline keyboards/linworks/fave87h/config.h:21]:
```suggestion
#define RGB_MATRIX_LED_COUNT 87
```
Conflict on this file - this define has been removed as the CLI can determine the LED count from the `g_led_config`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave87h/fave87h.c
Comment (fauxpark) [inline keyboards/linworks/fave87h/fave87h.c:1]: Please move the RGB config to keyboard.json. This file can then be deleted.

File: keyboards/linworks/fave87h/keymaps/default/keymap.c [hunk 11]
```diff
@@ -19,6 +19,7 @@ enum layers {
     _LAYER1,
 };
 
+
```
Comment (fauxpark) [inline keyboards/linworks/fave87h/keymaps/default/keymap.c:22]:
```suggestion
```

File: keyboards/linworks/fave87h/keymaps/via/keymap.c [hunk 12]
```diff
@@ -17,46 +17,27 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 enum layers {
     _LAYER0,
     _LAYER1,
-    _LAYER2,
-    _LAYER3,
 };
 
+
```
Comment (fauxpark) [inline keyboards/linworks/fave87h/keymaps/via/keymap.c:22]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by tzarc** [comment]
As of August 26, 2024, `qmk/qmk_firmware` is no longer accepting VIA-enabled keymaps as these have now transitioned to a repository under the VIA team's control.

As you've submitted a PR containing `via` or VIA-enabled keymap(s), this is your notice that they should be removed from this PR. You should now submit a secondary PR to the [VIA QMK Userspace](https://github.com/the-via/qmk_userspace_via) repository with your associated `via` or VIA-enabled keymaps instead.

**Pull request closed without merging**